src/main/java/hudson/plugins/scm/koji/KojiChangeLogSet.java

```diff
@@ -83,4 +81,60 @@ public User getAuthor() {
 
     }
 
+    private static List<Hyperlink> getListFromString(String string) {
+        List<Hyperlink> hyperlinks = new ArrayList<>();
+        hyperlinks.add(new Hyperlink(string));
+        return hyperlinks;
+    }
+
+    private static List<Hyperlink> getListFromRPMs(List<RPM> rpms) {
+        List<Hyperlink> hyperlinks = rpms.stream().filter(rpm -> rpm.hasUrl()).map(rpm -> new Hyperlink(rpm.toString(), rpm.getUrl())).collect(Collectors.toList());
+        return hyperlinks;
+    }
+
+    private static List<Hyperlink> getListFromUrl(URL url) {
+        List<Hyperlink> hyperlinks = new ArrayList<>();
+        if (url != null) {
+            if (RPM.Suffix.INSTANCE.endsWithSuffix(url.toString())) {
+                hyperlinks.add(new Hyperlink(new File(url.getPath()).getName(), url.toString()));
+            } else {
+                hyperlinks.add(new Hyperlink("Source file not found. You can search for it here.", url.toString()));
+            }
+        }
+        return hyperlinks;
+    }
+
+    public static class Hyperlink {
+
+        private final String displayedString;
+        private final String url;
+
+        public Hyperlink(String displayedString) {
+            this.displayedString = displayedString;
+            url = null;
+        }
+
+        public Hyperlink(String displayedString, String url) {
+            this.displayedString = displayedString;
+            this.url = url;
+        }
+
+        public String getDisplayedString() {
+            return displayedString;
+        }
+
+        public String getUrl() {
+            return url;
+        }
+
+        public boolean isContainingUrl() {
+            return url != null;
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format(displayedString);
```

Comment:
- Why call `String.format()` ?
- Please format this method same as others.
- Why you return just displayedString? Where is this called?

Reply:
Yes, I will fix this. It seems like this method isn't called anywhere, but it needs to be overridden.